src/main/java/bio/terra/service/configuration/ConfigFaultCounted.java

```diff
@@ -0,0 +1,144 @@
+            .enabled(isEnabled())
+            .counted(countedModel);
+
+        return new ConfigModel()
+            .configType(ConfigModel.ConfigTypeEnum.FAULT)
+            .name(getName())
+            .fault(faultCounted);
+    }
+
+    @Override
+    synchronized void setFaultConfig(ConfigFaultModel faultModel) {
+        if (faultModel.getCounted() == null) {
+            throw new ValidationException("Set of a counted fault requires a counted fault model");
+        }
+        countedModel = faultModel.getCounted();
+        initCounters();
+    }
+
+    @Override
+    synchronized void resetFaultConfig() {
+        countedModel = originalCountedModel;
+        initCounters();
+    }
+
+    @Override
+    synchronized boolean testInsertFault() {
+        if (!isEnabled()) {
+            return false;
+        }
+
+        if (!doneSkipping) {
+            skippedCounter++;
+            if (skippedCounter >= countedModel.getSkipFor()) {
+                doneSkipping = true;
+            }
+            return false;
+        }
+
+        if (!doneInserting) {
+            testCounter++;
+            switch (countedModel.getRateStyle()) {
+                case RANDOM:
+                    if (random.nextInt(100) < countedModel.getRate()) {
+                        logger.debug("Fault: random insert - insert=" + insertCounter + " test=" + testCounter);
+                        return doInsert();
+                    }
+                    break;
+
+                case FIXED:
+                    everyCounter++;
+                    logger.debug("Fault: fixed insert - every=" + every + " everyCounter=" + everyCounter);
+                    if (everyCounter >= every) {
+                        everyCounter = 0;
+                        return doInsert();
+                    }
+                    break;
+            }
+        }
+        return false;
+    }
+
+    private boolean doInsert() {
+        if (countedModel.getInsert() < 0) {
+            insertCounter++;
+            return true;
+        } else {
+            if (insertCounter < countedModel.getInsert()) {
+                insertCounter++;
+                return true;
+            } else {
+                doneInserting = true;
+                logger.debug("Fault: done inserting " + getConfigEnum() +
+                    "; inserted " + insertCounter + " of " +  testCounter + " tests");
+            }
+        }
+        return false;
+    }
+
+    private synchronized void initCounters() {
+        doneSkipping = (countedModel.getSkipFor() == 0);
+        skippedCounter = 0;
+        doneInserting = false;
+        insertCounter = 0;
+        testCounter = 0;
+        switch (countedModel.getRateStyle()) {
+            case RANDOM:
+                if (random == null) {
+                    random = new Random();
+                }
+                break;
+
+            case FIXED:
+                every = 100 / countedModel.getRate();
```

Comment:
Both of these are ints so there could be a loss of precision here. Can you switch this to `100.0`?

Reply:
Yes there is a loss of precision, but the `every` value is an integer; we are inserting the fault every integral number of times. Hard to implement every 3.33 times... You can generate these `every` integers: 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 14, 16, 20, 25, 33, 50, 100

If that is an insufficient set, then I should take a separate parameter for the fixed case that directly specified `every`.  What do you think?

Reply:
Or maybe split the fault into two different kinds: countedRandom and countedFixed?